client/watcher/watcher.go

```diff
@@ -74,8 +74,11 @@ func (w *Watcher) Watch() {
 			w.mu.Lock()
 			select {
 			case event := <-w.watcher.Events:
-				if event.Op&fsnotify.Write == fsnotify.Write {
-					w.UpdateBytes()
+				if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Create == fsnotify.Create {
```

Comment:
Is this code suppose to work exclusively with the file-writing code in `cli/api/client` is meant to be generic? In the latter case, it may still observe partial writes, of course. You can use the `inotify` methods in the `unix`package directly to listen for the `IN_CLOSE_WRITE` and the `IN_MOVED_TO` events in order to fix that. You also won't need atomic writes. And in the former case, the code does not need to handle the `fsnotify.Write` case.

Reply:
It's supposed to be somewhere in between. If the producer is using ioutils.WriteFile, a partial write will signal an error. (https://golang.org/src/io/ioutil/ioutil.go?s=2722:2780#L76)


Doing it this way will preserve compatibility with users that have extended this package in some way.

Reply:
The atomic writes are necessary for the PHP client, I think. It seems to be occasionally picking up a truncated file from ioutils.WriteFile. The node client also responds to every event without checking. We could fix all these, or just switch to atomic writes. I think the latter is simpler for consumers.